sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -63,6 +64,23 @@ case class InMemoryRelation(
     val batchStats: LongAccumulator = child.sqlContext.sparkContext.longAccumulator)
   extends logical.LeafNode with MultiInstanceRelation {
 
+  /**
+   * If true, get data from ColumnVector in ColumnarBatch, which are generally faster.
+   * If false, get data from UnsafeRow build from ColumnVector
+   */
+  private[columnar] val useColumnarBatches: Boolean = {
```

Comment:
I think this should belong to the read path. Whether we use columnar batch scan or not, the write path doesn't need to change.

Reply:
You are right. Previous implementation required to change execution path in the write path, too. After review comments, this flag is not referenced in `InMemoryRelation`. I will move this flag to `InMemoryTableScanExec`.